packages/modals/src/elements/Modal.tsx

```diff
@@ -119,7 +129,13 @@ export const Modal = React.forwardRef<HTMLDivElement, IModalProps>(
       restoreFocus
     });
 
-    useFocusVisible({ scope: modalRef });
+    useFocusVisible({ scope });
```

Comment:
From our offline talk, lets take a look at the `relativeDocument` option to try and get this to recognize the conditional ref.